### PR TITLE
Adjust to ArmCL 18.05

### PR DESCRIPTION
**Pull request by Chunosov**

### CODE REVIEW

File: program/mobilenets-armcl-opencl/armcl_graph_common.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#if defined(ARMCL_18_05_PLUS)
+  arm_compute::graph::backends::BackendRegistry::get()
+    .add_backend<arm_compute::graph::backends::CLDeviceBackend>(
+      arm_compute::graph::Target::CL);
+#endif
+}
+
+#if defined(ARMCL_18_05_PLUS)
+
+inline arm_compute::graph::DepthwiseConvolutionMethod str_to_dwsc_convolution_method(const char *method_name) {
```
Comment (psyhtest) [inline program/mobilenets-armcl-opencl/armcl_graph_common.h]: Takes some effort to translate `dwsc` :). Why not simply `depthwise`?

File: program/mobilenets-armcl-opencl/armcl_graph_common.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethod::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethod::DIRECT;
+  if (strcmp(method_name, "WINOGRAD") == 0) return arm_compute::graph::ConvolutionMethod::WINOGRAD;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::ConvolutionMethod>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethod get_convolution_method() {
+  auto method_name = getenv("CK_CONVOLUTION_METHOD");
+  if (!method_name) {
```
Comment (psyhtest) [inline program/mobilenets-armcl-opencl/armcl_graph_common.h]: A bit hard to read. How about:
```
if (method_name) {
    return str_to_convolution_method(method_name);
} else {
    if (arm_compute::CLScheduler::get().target() == arm_compute::GPUTarget::BIFROST) {
        return arm_compute::graph::ConvolutionMethod::DIRECT;
    } else {
        return arm_compute::graph::ConvolutionMethod::GEMM;
    }
}
```

File: program/mobilenets-armcl-opencl/armcl_graph_common.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+inline arm_compute::graph::ConvolutionMethodHint str_to_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethodHint::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethodHint::DIRECT;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::ConvolutionMethodHint>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethodHint get_convolution_method() {
```
Comment (psyhtest) [inline program/mobilenets-armcl-opencl/armcl_graph_common.h]: Isn't this the same as for the case `ArmCL >= 18.05`?

File: program/mobilenets-armcl-opencl/mobilenets.cpp [hunk 4]
```diff
@@ -141,6 +148,47 @@ class CKOutputAccessor : public ITensorAccessor {
...
 };
 
+std::string get_convolution_methods_file() {
+  auto filename = getenv("CK_CONVOLUTION_METHOD_FILE");
+  return filename ? std::string(filename) : std::string("conv_methods.txt");
+}
+
+std::vector<int> load_convolution_methods() {
+  const int convolutions_count = 15;
+  const int defaut_method = static_cast<int>(get_convolution_method());
```
Comment (psyhtest) [inline program/mobilenets-armcl-opencl/mobilenets.cpp]: `default`

File: program/mobilenets-armcl-opencl/armcl_graph_common.h (hunk 1 above)
Comment (Chunosov) [reply program/mobilenets-armcl-opencl/armcl_graph_common.h]: This method is no more needed, removed.

Comment (Chunosov) [reply program/mobilenets-armcl-opencl/armcl_graph_common.h]: But the name was from ARM's `get_dwsc_node`

File: program/mobilenets-armcl-opencl/armcl_graph_common.h (hunk 2 above)
Comment (Chunosov) [reply program/mobilenets-armcl-opencl/armcl_graph_common.h]: Fixed. But I personally don't like `else` when `if` returns. So if you don't insist... :)

File: program/mobilenets-armcl-opencl/armcl_graph_common.h (hunk 3 above)
Comment (Chunosov) [reply program/mobilenets-armcl-opencl/armcl_graph_common.h]: When `ArmCL >= 18.05` then `arm_compute::graph::ConvolutionMethod` is used, but not `arm_compute::graph::ConvolutionMethodHint`

File: program/mobilenets-armcl-opencl/armcl_graph_common.h [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    return arm_compute::graph::ConvolutionMethodHint::GEMM;
+
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethodHint::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethodHint::DIRECT;
+  
+  // Try to get convolution method as integer value.
+  switch (atoi(method_name)) {
+    case 0: return arm_compute::graph::ConvolutionMethodHint::GEMM;
+    case 1: arm_compute::graph::ConvolutionMethodHint::DIRECT;
```
Comment (psyhtest) [inline program/mobilenets-armcl-opencl/armcl_graph_common.h]: return

**Pull request merged**